### PR TITLE
fix: make ErrorBase always capture stack trace

### PR DESCRIPTION
**Pull request by lukekarrys**
fix: allow HttpErrorBase to take headers object

### CODE REVIEW

File: lib/errors.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 'use strict'
 
-const url = require('url')
+const { URL } = require('url')
```
Comment (wraithgar) [inline lib/errors.js:3]:
```suggestion
const { URL } = require('node:url')
```

File: lib/errors.js [hunk 2]
```diff
@@ -24,16 +24,16 @@ class HttpErrorBase extends Error {
   constructor (method, res, body, spec) {
     super()
     this.name = this.constructor.name
-    this.headers = res.headers.raw()
+    this.headers = typeof res.headers.raw === 'function' ? res.headers.raw() : res.headers
```
Comment (wraithgar) [inline lib/errors.js:27]: What caused this change?

**Comment by wraithgar** [comment]
headers.raw change will need a test and I'd love to know what cause that to be needed.

File: lib/errors.js (hunk 2 above)
Comment (lukekarrys) [reply lib/errors.js:27]: https://github.com/npm/npm-profile/blob/lk/abort-signal/lib/index.js#L73

**Comment by lukekarrys** [comment]
Added a test for `headers.raw`. These errors are used in `npm-profile` and there's one spot there a headers object was given a `raw` function just to return the object like `headers: { raw: () => er.headers }`

File: lib/errors.js [hunk 3]
```diff
@@ -24,16 +24,17 @@ class HttpErrorBase extends Error {
   constructor (method, res, body, spec) {
     super()
     this.name = this.constructor.name
-    this.headers = res.headers.raw()
+    console.log(res.headers)
```
Comment (wraithgar) [inline lib/errors.js:27]: console.log

Comment (lukekarrys) [inline lib/errors.js:27]:
```suggestion
```

**Review by wraithgar — APPROVED** [review]

**Pull request merged**